src/lib/coda_lib/coda_lib.ml

```diff
@@ -658,10 +664,34 @@ module Make (Inputs : Inputs_intf) = struct
             let frontier_broadcast_pipe_r, frontier_broadcast_pipe_w =
               Broadcast_pipe.create (Some transition_frontier)
             in
+            let mplus ma mb = if Option.is_some ma then ma else mb in
             let%bind net =
               Net.create config.net_config
-                ~get_staged_ledger_aux_at_hash:(fun _hash ->
-                  failwith "shouldn't be necessary right now?" )
+                ~get_staged_ledger_aux_and_pending_coinbases_at_hash:
+                  (fun enveloped_hash ->
+                  let hash = Envelope.Incoming.data enveloped_hash in
+                  Deferred.return
+                  @@
+                  let open Option.Let_syntax in
+                  let%bind frontier =
+                    Broadcast_pipe.Reader.peek frontier_broadcast_pipe_r
+                  in
+                  let%map breadcrumb =
+                    mplus
+                      (Transition_frontier.find frontier hash)
+                      (Transition_frontier.find_in_root_history frontier hash)
+                  in
+                  let staged_ledger =
+                    Transition_frontier.Breadcrumb.staged_ledger breadcrumb
+                  in
+                  let scan_state = Staged_ledger.scan_state staged_ledger in
+                  let merkle_root =
```

Comment:
We don't need to ask the network for the merkle_root `merkle_root` if `Syncable_ledger` stores the root external_transition.